chore: fix context definition

ntn-x2 opened the pull request.

rflechtner commented on README.md:
```diff
@@ -83,26 +83,48 @@ The same process is followed for the asset identifier.
+{
+  "@protected": true,
+  "id": "@id",
+  "asset-did-spec": "https://github.com/KILTprotocol/spec-asset-did/blob/main/README.md",
```
Did you check what the predicates expand to? Not sure, but you might need to add a `#` to make sure the result is a valid URL

rflechtner commented on README.md:
```diff
@@ -83,26 +83,48 @@ The same process is followed for the asset identifier.
+    "namespace": "asset-did-spec:asset-namespace",
+    "reference": "asset-did-spec:asset-reference",
+    "identifier": "asset-did-spec:asset-identifier"
+  },
```
this is one `}` too many

rflechtner commented on README.md:
```diff
@@ -83,26 +83,48 @@ The same process is followed for the asset identifier.
+  "id": "@id",
+  "asset-did-spec": "https://github.com/KILTprotocol/spec-asset-did/blob/main/README.md",
+  "chain": {
+    "@id": "asset-did-spec:chain",
```
you could add ` "@type":"@id",` for completeness, tells the processor that we expect the value of `chain` to be an IRI. Just covers the case where you'd have `"chain": "eip155:0"` in your document so yeah, just for completeness.

rflechtner commented on README.md:
```diff
@@ -83,26 +83,48 @@ The same process is followed for the asset identifier.
   "@context": [
     "https://www.w3.org/ns/did/v1",
-    "https://kilt.io/asset-did-context.json"
+    "https://github.com/KILTprotocol/spec-asset-did/blob/main/README.md"
```
the context url is expected to return the context document, i.e. JSON, so linking to a markdown file doesn't make too much sense here. I think you can keep it as it was (or use an ipfs URI)

rflechtner reviewed (commented):
I like it! I think with that structure and context we have a very solid base for representing the information contained in a CAIP-19 identifier in a LD fashion. We could explain better what the predicates mean exactly (I think that is the point of predicates being resolvable URLs in the LD approach), but for starters we need to defined the structure and I think we're on to something here. I can already see how we can build on top of this, e.g.
- The `alsoKnownAs` field could contain the standard form of the CAIP-19 identifier
- We could define chain `@type`s such as `EthereumBased`, `SubstrateBased`
- We could define `@type`s or a predicate which tell you whether you are looking at an asset class or an asset instance
- etc. etc.

There's also a larger discussion to be had regarding how we imagine the fit of CAIP-19, a referencing framework, and DIDs, an SSI framework, but I'm bracketing that here bc I don't think it's the right place.

ntn-x2 commented:
@rflechtner I addressed your points in https://github.com/KILTprotocol/spec-asset-did/pull/1/commits/04af93d8b7ce17067e410b5e8baacb95924603ae, thanks a lot for the suggestions!

I decided to actually go for an embedded context in the example resolution, and I think I will mirror that in the SDK as well, since it actually does make sense to me, that we ship a new context whenever we change something in the resolver anyway. Maybe we should do the same for KILT DIDs as well, what do you think?

Anyway, this is a screenshot of how the context is evaluated now with the example data, which looks good enough to me.

<img width="1037" alt="Screenshot 2023-02-23 at 17 27 03" src="https://user-images.githubusercontent.com/6704504/220968932-794a550c-baeb-40d4-8c52-c540928d7278.png">

rflechtner commented on README.md:
```diff
@@ -83,26 +83,61 @@ The same process is followed for the asset identifier.
-    "https://kilt.io/asset-did-context.json"
+    {
+      "@protected": true,
+      "asset-did-spec": "https://github.com/KILTprotocol/spec-asset-did/blob/main/README.md",
```
This is still broken I think. Try adding the hash in the playground and observe how the expanded result changes 
```suggestion
      "asset-did-spec": "https://github.com/KILTprotocol/spec-asset-did/blob/main/README.md#",
```

rflechtner reviewed (changes requested):
Nice, but one of my comments (https://github.com/KILTprotocol/spec-asset-did/pull/1/files#r1112908032) still needs to be addressed before we can go ahead. I think this breaks JSON-LD processing.

> I decided to actually go for an embedded context in the example resolution, and I think I will mirror that in the SDK as well, since it actually does make sense to me, that we ship a new context whenever we change something in the resolver anyway. Maybe we should do the same for KILT DIDs as well, what do you think?

I'm not too sure about the embedding; all this extra data may look confusing to somebody who doesn't know what JSON-LD is. And from an efficiency standpoint it's better to fetch a context definition once instead of including with every did resolution of course. But I'm not totally against it.
For KILT DIDs, the context definition internally links to yet another context definition, so I'm not sure we gain a lot by embedding it; and the same arguments about size and complexity apply.

rflechtner commented:
> Anyway, this is a screenshot of how the context is evaluated now with the example data, which looks good enough to me.
> 
> <img alt="Screenshot 2023-02-23 at 17 27 03" width="1037" src="https://user-images.githubusercontent.com/6704504/220968932-794a550c-baeb-40d4-8c52-c540928d7278.png">

BTW I think there is a bug in this viewer; it should look like what I pasted below but you only get this if you use `@id` instead of `id` (which are supposed to be synonyms?). As far as I can tell the table view gives you identical statements for both variants.

![Screenshot 2023-02-28 at 13 51 54](https://user-images.githubusercontent.com/39338561/221859430-08cec5e6-ecbc-420f-8b6b-defba5214fa9.png)
